### DISCUSSION TITLE
Determining the optimal number of states for a given model

### DISCUSSION THREAD

**Discussion by SiemScholman** [vidaurre/glhmm#59] [Q&A] [answered]
Dear Diego,

My name is Siem Scholman, I'm a Master's student studying neuroscience at Utrecht University in The Netherlands and I'm using the glhmm package to fit a standard Gaussian HMM to EEG data for my mental chronometry study on decision making. Do you know of a way to compare the fits of multiple models with varying values for K? Or in other words, how do I select the optimal number of states for my model?

**Comment by vidaurre** [comment, marked as answer]
Thanks Siem,

The most common way to choose the number of states is the free energy. But given that EEG comprises a lot of data, generally, the free energy will ask for too many states. A typically more useful metric is based on half splits: you divide the data in 2 random sets and check if the states are comparable across the two splits. If it's not, you should probably ask for fewer. 

Nonetheless the number of states often comes determined by your research question. Sometimes you need more, sometimes you need fewer.

Hope that helps!
